public/components/health-check/health-check.tsx

```diff
@@ -247,6 +239,40 @@ export class HealthCheck extends Component {
         }
     }
 
+    /**
+     * This check if the pattern exist then create if not
+     * @param pattern string
+     */
+    async checkSupportPattern(pattern, itemId) {
+        let results = this.state.results;
+        let errors = this.state.errors;
+        const result = await SavedObject.existsIndexPattern(pattern);
+        if (!result.data) {
+            const toast = getToasts().addWarning(`${pattern} index pattern was not found and it will be created`)
+            const fields = await SavedObject.getIndicesFields(pattern);
+            try {
+                await SavedObject.createSavedObject(
+                    'index-pattern',
+                    pattern,
+                    {
+                    attributes: {
+                        title: pattern,
+                        timeFieldName: 'timestamp'
+                    }
+                    },
+                    fields
+                );
+                getToasts().remove(toast.id);
+                getToasts().addSuccess(`${pattern} index pattern created successfully`)
+                results[itemId].description = <span><EuiIcon type="check" color="secondary" ></EuiIcon> Ready</span>;                
+            } catch (error) {
+                getToasts().remove(toast.id);
+                errors.push(`Error trying to create ${pattern} index pattern: ${error.message}`);
+                results[itemId].description = <span><EuiIcon type="alert" color="danger" ></EuiIcon> Error</span>;
+            }
+        }
+    }
```

Comment:
If you mute the `results` variable that is in the component's state. You should set the state after the mutation, like this:

```
this.setState( { results } );
```

You could do this after the `try/catch` block.

Reply:
I missed that thanks